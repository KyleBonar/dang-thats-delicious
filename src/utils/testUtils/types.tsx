export interface MockLink {
  text: string;
  to: string;
  target: "_blank" | "_self";
}
