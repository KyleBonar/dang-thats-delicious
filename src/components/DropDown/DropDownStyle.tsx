import styled from "../../theme/styled-components";

export const SelectContainer = styled.div`
  width: 100%;
  border: 1px solid #e1e1e1;
  color: #4b4b4b;
  display: inline-block;
  position: relative;
  background-color: #f5f5f5;

  &:after {
    position: absolute;
    top: 55%;
    transform: translateY(-50%);
    content: url("../../../../../../images/icons/chevron-down.svg");
    pointer-events: none;
    right: 15px;
  }
`;

export const StyledSelect = styled.select`
  width: 100%;
  background-color: transparent;
  appearance: none;
  border: 0;
  cursor: pointer;
  display: inline-block;
  position: relative;
  transition: background-color 0.3s, color 0.3s, border 0.3s;
  vertical-align: middle;
  height: 58px;
  padding: 0px 40px 0 15px;
  font-size: 1rem;
`;
StyledSelect.displayName = "StyledSelect"; // useful for testing
