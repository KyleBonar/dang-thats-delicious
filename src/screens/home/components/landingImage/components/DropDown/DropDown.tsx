import * as React from "react";
import * as shortid from "shortid";

import styled from "../../../../../../theme/styled-components";

const SelectContainer = styled.div`
  border: 1px solid #e1e1e1;
  border-radius: 4px;
  color: #4b4b4b;
  display: inline-block;
  position: relative;
  background-color: #f5f5f5;
  border-color: #eaeaea;
  border-bottom: 0;
  border-bottom-right-radius: 0;
  border-left: 0;
  border-top: 0;
  border-top-right-radius: 0;

  &:after {
    position: absolute;
    top: 55%;
    transform: translateY(-50%);
    content: url("../../../../../../images/icons/chevron-down.svg");
    pointer-events: none;
    right: 15px;
  }
`;

const Select = styled.select`
  background-color: transparent;
  appearance: none;
  border: 0;
  cursor: pointer;
  display: inline-block;
  min-height: 30px;
  position: relative;
  transition: background-color 0.3s, color 0.3s, border 0.3s;
  vertical-align: middle;
  height: 58px;
  padding: 0px 40px 0 15px;
  font-size: 1rem;
`;

interface DropDownProps {
  id?: string;
  showLabel?: boolean;
  label?: string;
  backgroundColor?: string;
  textColor?: string;
  options: string[];
}

const DropDown: React.SFC<DropDownProps> = props => {
  const SelectOption = (val: string): React.ReactElement<HTMLOptionElement> => {
    return (
      <option key={shortid.generate()} value={val}>
        {val}
      </option>
    );
  };

  return (
    <SelectContainer>
      {props.showLabel &&
        props.label && <label htmlFor={props.id}>{props.label}</label>}
      <Select id={props.id}>{props.options.map(SelectOption)}</Select>
    </SelectContainer>
  );
};

DropDown.defaultProps = {
  id: shortid.generate(),
  showLabel: false
};

export default DropDown;
