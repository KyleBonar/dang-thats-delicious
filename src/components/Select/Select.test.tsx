import * as React from "react";
import * as enzyme from "enzyme";

import { casual, ITERATION_SIZE } from "../../utils/testUtils/testUtils";
import DropDown, { SelectProps } from "./Select";

const getRandomValFromArr = (arr: string[]) => {
  // get rnadom value from array if possible
  const randomIndexInArray = casual.integer(0, arr.length);
  // console.log(randomIndexInArray, arr);

  return arr[randomIndexInArray];
};

const fakeSelect = (): SelectProps => {
  // generate options
  const options = casual.array_of_words();

  // get selected value if possible
  const selectedValue = casual.random_element([
    undefined,
    getRandomValFromArr(options)
  ]);

  return {
    id: casual.random_element([undefined, casual.uuid]),
    options,
    selectedValue,
    name: casual.random_element([undefined, casual.uuid]),
    className: casual.random_element([undefined, casual.string]),
    showLabel: casual.random_element([undefined, casual.boolean]),
    label: casual.random_element([undefined, casual.string]),
    required: casual.random_element([undefined, casual.boolean]),
    onSelect: jest.fn()
  };
};

const mockDropDowns = new Array(ITERATION_SIZE).fill(null).map(fakeSelect);

describe("<Select />", () => {
  it("renders", () => {
    mockDropDowns.forEach(mockDropDown => {
      const wrapper = enzyme.shallow(<DropDown {...mockDropDown} />);
      expect(wrapper).toBeTruthy();
    });
  });

  it("matches snapshot", () => {
    mockDropDowns.forEach(mockDropDown => {
      const wrapper = enzyme.shallow(<DropDown {...mockDropDown} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  it("renders correct number of options", () => {
    mockDropDowns.forEach(mockDropDown => {
      const wrapper = enzyme.shallow(<DropDown {...mockDropDown} />);

      expect(wrapper.find("option").length).toEqual(
        mockDropDown.options.length
      );
    });
  });

  it("renders select tag with expected selectedValue value", () => {
    mockDropDowns.forEach(mockDropDown => {
      const wrapper = enzyme.shallow(<DropDown {...mockDropDown} />);

      expect(wrapper.find("select").prop("value")).toEqual(
        mockDropDown.selectedValue
      );
    });
  });

  it("calls onSelect when component used", () => {
    mockDropDowns.forEach(mockDropDown => {
      const wrapper = enzyme.shallow(<DropDown {...mockDropDown} />);

      const select = wrapper.find("select");

      select.simulate("change");
      expect(mockDropDown.onSelect).toHaveBeenCalledTimes(1);

      select.simulate("change");
      expect(mockDropDown.onSelect).toHaveBeenCalledTimes(2);
    });
  });

  it("adds className to parent div", () => {
    mockDropDowns.forEach(mockDropDown => {
      const wrapper = enzyme.shallow(<DropDown {...mockDropDown} />);

      if (!mockDropDown.className) return;

      expect(wrapper.find("div").first().prop("className")).toContain(
        mockDropDown.className
      );
    });
  });

  it("renders label element when showLabel is true", () => {
    mockDropDowns.forEach(mockDropDown => {
      const wrapper = enzyme.shallow(<DropDown {...mockDropDown} />);

      if (!mockDropDown.showLabel) return;

      expect(wrapper.find("Label")).toBeTruthy();
    });
  });
});
