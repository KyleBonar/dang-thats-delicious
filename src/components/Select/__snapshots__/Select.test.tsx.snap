// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> matches snapshot 1`] = `
<div>
  <div>
    <select
      name="e2368339-2e09-4049-8fd0-2e808d9d8109"
      onChange={[MockFunction]}
      value="corporis"
    >
      <option
        key="0-sequi"
        value="sequi"
      >
        sequi
      </option>
      <option
        key="1-facilis"
        value="facilis"
      >
        facilis
      </option>
      <option
        key="2-corporis"
        value="corporis"
      >
        corporis
      </option>
      <option
        key="3-sed"
        value="sed"
      >
        sed
      </option>
      <option
        key="4-veritatis"
        value="veritatis"
      >
        veritatis
      </option>
      <option
        key="5-quidem"
        value="quidem"
      >
        quidem
      </option>
      <option
        key="6-praesentium"
        value="praesentium"
      >
        praesentium
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 2`] = `
<div
  className="in sunt occaecati nulla aut sunt est"
>
  <div>
    <select
      onChange={[MockFunction]}
      value="omnis"
    >
      <option
        key="0-itaque"
        value="itaque"
      >
        itaque
      </option>
      <option
        key="1-omnis"
        value="omnis"
      >
        omnis
      </option>
      <option
        key="2-debitis"
        value="debitis"
      >
        debitis
      </option>
      <option
        key="3-rem"
        value="rem"
      >
        rem
      </option>
      <option
        key="4-similique"
        value="similique"
      >
        similique
      </option>
      <option
        key="5-temporibus"
        value="temporibus"
      >
        temporibus
      </option>
      <option
        key="6-aliquam"
        value="aliquam"
      >
        aliquam
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 3`] = `
<div>
  <div>
    <select
      name="7bc68196-f8d1-43d3-ad97-c60ec9cfc616"
      onChange={[MockFunction]}
    >
      <option
        key="0-assumenda"
        value="assumenda"
      >
        assumenda
      </option>
      <option
        key="1-quis"
        value="quis"
      >
        quis
      </option>
      <option
        key="2-quae"
        value="quae"
      >
        quae
      </option>
      <option
        key="3-iusto"
        value="iusto"
      >
        iusto
      </option>
      <option
        key="4-sapiente"
        value="sapiente"
      >
        sapiente
      </option>
      <option
        key="5-nostrum"
        value="nostrum"
      >
        nostrum
      </option>
      <option
        key="6-aut"
        value="aut"
      >
        aut
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 4`] = `
<div
  className="ab assumenda animi non quis quis nobis"
>
  <div>
    <select
      id="4a67b9b1-ab9c-4f94-aabc-26e8a2928077"
      name="ba28c927-828a-4031-9860-d06ee101ad55"
      onChange={[MockFunction]}
      value="sint"
    >
      <option
        key="0-aut"
        value="aut"
      >
        aut
      </option>
      <option
        key="1-autem"
        value="autem"
      >
        autem
      </option>
      <option
        key="2-ut"
        value="ut"
      >
        ut
      </option>
      <option
        key="3-ullam"
        value="ullam"
      >
        ullam
      </option>
      <option
        key="4-et"
        value="et"
      >
        et
      </option>
      <option
        key="5-possimus"
        value="possimus"
      >
        possimus
      </option>
      <option
        key="6-sint"
        value="sint"
      >
        sint
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 5`] = `
<div
  className="expedita omnis sit vitae consequuntur non sit"
>
  <div>
    <select
      onChange={[MockFunction]}
    >
      <option
        key="0-ut"
        value="ut"
      >
        ut
      </option>
      <option
        key="1-dignissimos"
        value="dignissimos"
      >
        dignissimos
      </option>
      <option
        key="2-aspernatur"
        value="aspernatur"
      >
        aspernatur
      </option>
      <option
        key="3-voluptate"
        value="voluptate"
      >
        voluptate
      </option>
      <option
        key="4-voluptatem"
        value="voluptatem"
      >
        voluptatem
      </option>
      <option
        key="5-est"
        value="est"
      >
        est
      </option>
      <option
        key="6-eius"
        value="eius"
      >
        eius
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 6`] = `
<div>
  <div>
    <select
      name="f67e1947-6f3c-41b3-a1d0-dbb4e0c1e361"
      onChange={[MockFunction]}
    >
      <option
        key="0-sed"
        value="sed"
      >
        sed
      </option>
      <option
        key="1-nobis"
        value="nobis"
      >
        nobis
      </option>
      <option
        key="2-excepturi"
        value="excepturi"
      >
        excepturi
      </option>
      <option
        key="3-saepe"
        value="saepe"
      >
        saepe
      </option>
      <option
        key="4-autem"
        value="autem"
      >
        autem
      </option>
      <option
        key="5-quis"
        value="quis"
      >
        quis
      </option>
      <option
        key="6-molestiae"
        value="molestiae"
      >
        molestiae
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 7`] = `
<div
  className="ut distinctio molestias placeat exercitationem unde minus"
>
  <div>
    <select
      name="6fd3a9e0-61af-48f0-9fde-3b658877621f"
      onChange={[MockFunction]}
    >
      <option
        key="0-voluptatibus"
        value="voluptatibus"
      >
        voluptatibus
      </option>
      <option
        key="1-reiciendis"
        value="reiciendis"
      >
        reiciendis
      </option>
      <option
        key="2-ut"
        value="ut"
      >
        ut
      </option>
      <option
        key="3-eos"
        value="eos"
      >
        eos
      </option>
      <option
        key="4-commodi"
        value="commodi"
      >
        commodi
      </option>
      <option
        key="5-at"
        value="at"
      >
        at
      </option>
      <option
        key="6-impedit"
        value="impedit"
      >
        impedit
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 8`] = `
<div
  className="ea quia sunt est aut et sunt"
>
  <div>
    <select
      onChange={[MockFunction]}
      value="voluptatem"
    >
      <option
        key="0-ipsa"
        value="ipsa"
      >
        ipsa
      </option>
      <option
        key="1-odio"
        value="odio"
      >
        odio
      </option>
      <option
        key="2-sit"
        value="sit"
      >
        sit
      </option>
      <option
        key="3-laboriosam"
        value="laboriosam"
      >
        laboriosam
      </option>
      <option
        key="4-quis"
        value="quis"
      >
        quis
      </option>
      <option
        key="5-voluptatem"
        value="voluptatem"
      >
        voluptatem
      </option>
      <option
        key="6-quod"
        value="quod"
      >
        quod
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 9`] = `
<div>
  <div>
    <select
      id="38ee790e-d9de-4d29-835f-1a7ae53c8fe7"
      onChange={[MockFunction]}
    >
      <option
        key="0-quia"
        value="quia"
      >
        quia
      </option>
      <option
        key="1-voluptate"
        value="voluptate"
      >
        voluptate
      </option>
      <option
        key="2-optio"
        value="optio"
      >
        optio
      </option>
      <option
        key="3-suscipit"
        value="suscipit"
      >
        suscipit
      </option>
      <option
        key="4-delectus"
        value="delectus"
      >
        delectus
      </option>
      <option
        key="5-aliquid"
        value="aliquid"
      >
        aliquid
      </option>
      <option
        key="6-et"
        value="et"
      >
        et
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 10`] = `
<div>
  <div>
    <select
      id="b53e8ee5-873a-4b54-b488-945ab7610b95"
      onChange={[MockFunction]}
      value="repellat"
    >
      <option
        key="0-asperiores"
        value="asperiores"
      >
        asperiores
      </option>
      <option
        key="1-amet"
        value="amet"
      >
        amet
      </option>
      <option
        key="2-et"
        value="et"
      >
        et
      </option>
      <option
        key="3-quos"
        value="quos"
      >
        quos
      </option>
      <option
        key="4-repellat"
        value="repellat"
      >
        repellat
      </option>
      <option
        key="5-distinctio"
        value="distinctio"
      >
        distinctio
      </option>
      <option
        key="6-libero"
        value="libero"
      >
        libero
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 11`] = `
<div>
  <div>
    <select
      name="780c6179-b731-4e8f-ab41-5dfc89bbb55d"
      onChange={[MockFunction]}
    >
      <option
        key="0-quisquam"
        value="quisquam"
      >
        quisquam
      </option>
      <option
        key="1-quia"
        value="quia"
      >
        quia
      </option>
      <option
        key="2-blanditiis"
        value="blanditiis"
      >
        blanditiis
      </option>
      <option
        key="3-qui"
        value="qui"
      >
        qui
      </option>
      <option
        key="4-sed"
        value="sed"
      >
        sed
      </option>
      <option
        key="5-a"
        value="a"
      >
        a
      </option>
      <option
        key="6-dolores"
        value="dolores"
      >
        dolores
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 12`] = `
<div
  className="repellendus quaerat similique autem doloribus et voluptatem"
>
  <div>
    <select
      id="66643f02-e4a5-472a-b0fa-909cdeb7d467"
      onChange={[MockFunction]}
      value="est"
    >
      <option
        key="0-et"
        value="et"
      >
        et
      </option>
      <option
        key="1-aperiam"
        value="aperiam"
      >
        aperiam
      </option>
      <option
        key="2-quia"
        value="quia"
      >
        quia
      </option>
      <option
        key="3-est"
        value="est"
      >
        est
      </option>
      <option
        key="4-et"
        value="et"
      >
        et
      </option>
      <option
        key="5-suscipit"
        value="suscipit"
      >
        suscipit
      </option>
      <option
        key="6-ipsa"
        value="ipsa"
      >
        ipsa
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 13`] = `
<div>
  <Label
    htmlFor="6db8a202-4179-4f01-8642-da8cc8612e99"
  >
    dolore aut iste eos perspiciatis eius quod
    *
  </Label>
  <div>
    <select
      id="6db8a202-4179-4f01-8642-da8cc8612e99"
      onChange={[MockFunction]}
      value="aliquid"
    >
      <option
        key="0-assumenda"
        value="assumenda"
      >
        assumenda
      </option>
      <option
        key="1-nihil"
        value="nihil"
      >
        nihil
      </option>
      <option
        key="2-ad"
        value="ad"
      >
        ad
      </option>
      <option
        key="3-ut"
        value="ut"
      >
        ut
      </option>
      <option
        key="4-maxime"
        value="maxime"
      >
        maxime
      </option>
      <option
        key="5-aliquid"
        value="aliquid"
      >
        aliquid
      </option>
      <option
        key="6-sint"
        value="sint"
      >
        sint
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 14`] = `
<div>
  <div>
    <select
      id="84f9398f-77b2-4214-a391-bf1cf8226b95"
      name="4f2bef25-d9a5-494d-9daa-2874f23ba253"
      onChange={[MockFunction]}
      value="sed"
    >
      <option
        key="0-est"
        value="est"
      >
        est
      </option>
      <option
        key="1-ab"
        value="ab"
      >
        ab
      </option>
      <option
        key="2-quia"
        value="quia"
      >
        quia
      </option>
      <option
        key="3-ab"
        value="ab"
      >
        ab
      </option>
      <option
        key="4-sed"
        value="sed"
      >
        sed
      </option>
      <option
        key="5-voluptas"
        value="voluptas"
      >
        voluptas
      </option>
      <option
        key="6-ullam"
        value="ullam"
      >
        ullam
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 15`] = `
<div>
  <div>
    <select
      onChange={[MockFunction]}
    >
      <option
        key="0-iure"
        value="iure"
      >
        iure
      </option>
      <option
        key="1-iste"
        value="iste"
      >
        iste
      </option>
      <option
        key="2-aut"
        value="aut"
      >
        aut
      </option>
      <option
        key="3-ab"
        value="ab"
      >
        ab
      </option>
      <option
        key="4-laborum"
        value="laborum"
      >
        laborum
      </option>
      <option
        key="5-sint"
        value="sint"
      >
        sint
      </option>
      <option
        key="6-qui"
        value="qui"
      >
        qui
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 16`] = `
<div>
  <div>
    <select
      name="74bbc9a7-c810-4b38-960f-6652aae84085"
      onChange={[MockFunction]}
      value="est"
    >
      <option
        key="0-laboriosam"
        value="laboriosam"
      >
        laboriosam
      </option>
      <option
        key="1-sed"
        value="sed"
      >
        sed
      </option>
      <option
        key="2-est"
        value="est"
      >
        est
      </option>
      <option
        key="3-et"
        value="et"
      >
        et
      </option>
      <option
        key="4-sapiente"
        value="sapiente"
      >
        sapiente
      </option>
      <option
        key="5-ipsa"
        value="ipsa"
      >
        ipsa
      </option>
      <option
        key="6-aut"
        value="aut"
      >
        aut
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 17`] = `
<div>
  <div>
    <select
      id="b91869a7-5319-48c5-bc97-f0345cf64a08"
      name="2e9d2a1d-5a43-4630-a0e4-fd88335c6c3a"
      onChange={[MockFunction]}
    >
      <option
        key="0-fuga"
        value="fuga"
      >
        fuga
      </option>
      <option
        key="1-adipisci"
        value="adipisci"
      >
        adipisci
      </option>
      <option
        key="2-saepe"
        value="saepe"
      >
        saepe
      </option>
      <option
        key="3-quam"
        value="quam"
      >
        quam
      </option>
      <option
        key="4-quia"
        value="quia"
      >
        quia
      </option>
      <option
        key="5-veritatis"
        value="veritatis"
      >
        veritatis
      </option>
      <option
        key="6-qui"
        value="qui"
      >
        qui
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 18`] = `
<div>
  <div>
    <select
      id="0e71168e-9dfd-42ab-bc8d-33c406ccfa6f"
      onChange={[MockFunction]}
      value="quia"
    >
      <option
        key="0-ducimus"
        value="ducimus"
      >
        ducimus
      </option>
      <option
        key="1-est"
        value="est"
      >
        est
      </option>
      <option
        key="2-vel"
        value="vel"
      >
        vel
      </option>
      <option
        key="3-quia"
        value="quia"
      >
        quia
      </option>
      <option
        key="4-fugit"
        value="fugit"
      >
        fugit
      </option>
      <option
        key="5-totam"
        value="totam"
      >
        totam
      </option>
      <option
        key="6-quia"
        value="quia"
      >
        quia
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 19`] = `
<div>
  <div>
    <select
      name="5d48ba9b-e8cb-4f9c-8d01-9dfea73f5ffe"
      onChange={[MockFunction]}
      value="consequatur"
    >
      <option
        key="0-adipisci"
        value="adipisci"
      >
        adipisci
      </option>
      <option
        key="1-ab"
        value="ab"
      >
        ab
      </option>
      <option
        key="2-rerum"
        value="rerum"
      >
        rerum
      </option>
      <option
        key="3-quos"
        value="quos"
      >
        quos
      </option>
      <option
        key="4-consequatur"
        value="consequatur"
      >
        consequatur
      </option>
      <option
        key="5-assumenda"
        value="assumenda"
      >
        assumenda
      </option>
      <option
        key="6-odio"
        value="odio"
      >
        odio
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 20`] = `
<div
  className="sit illo consequatur itaque vero repellendus ex"
>
  <div>
    <select
      name="15a1f952-99a4-49dd-a946-10735336ccf8"
      onChange={[MockFunction]}
      value="velit"
    >
      <option
        key="0-nihil"
        value="nihil"
      >
        nihil
      </option>
      <option
        key="1-illum"
        value="illum"
      >
        illum
      </option>
      <option
        key="2-ut"
        value="ut"
      >
        ut
      </option>
      <option
        key="3-nobis"
        value="nobis"
      >
        nobis
      </option>
      <option
        key="4-velit"
        value="velit"
      >
        velit
      </option>
      <option
        key="5-velit"
        value="velit"
      >
        velit
      </option>
      <option
        key="6-perspiciatis"
        value="perspiciatis"
      >
        perspiciatis
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 21`] = `
<div
  className="dolores dolore velit ipsum ratione expedita modi"
>
  <div>
    <select
      id="ee438c5d-7a89-40c8-8a28-a8c0d662fecd"
      name="073a00b2-654a-487d-8e61-b434a1b9e31e"
      onChange={[MockFunction]}
      value="quaerat"
    >
      <option
        key="0-voluptates"
        value="voluptates"
      >
        voluptates
      </option>
      <option
        key="1-similique"
        value="similique"
      >
        similique
      </option>
      <option
        key="2-laboriosam"
        value="laboriosam"
      >
        laboriosam
      </option>
      <option
        key="3-molestiae"
        value="molestiae"
      >
        molestiae
      </option>
      <option
        key="4-modi"
        value="modi"
      >
        modi
      </option>
      <option
        key="5-expedita"
        value="expedita"
      >
        expedita
      </option>
      <option
        key="6-quaerat"
        value="quaerat"
      >
        quaerat
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 22`] = `
<div
  className="qui sint enim ratione cupiditate cumque suscipit"
>
  <div>
    <select
      name="0493c245-b16a-43b0-a23f-5725b46c8986"
      onChange={[MockFunction]}
      value="voluptatum"
    >
      <option
        key="0-aut"
        value="aut"
      >
        aut
      </option>
      <option
        key="1-et"
        value="et"
      >
        et
      </option>
      <option
        key="2-quis"
        value="quis"
      >
        quis
      </option>
      <option
        key="3-voluptatum"
        value="voluptatum"
      >
        voluptatum
      </option>
      <option
        key="4-est"
        value="est"
      >
        est
      </option>
      <option
        key="5-eos"
        value="eos"
      >
        eos
      </option>
      <option
        key="6-vero"
        value="vero"
      >
        vero
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 23`] = `
<div
  className="minus et consequatur veritatis animi autem necessitatibus"
>
  <div>
    <select
      name="62ae4d44-45ab-424e-aafa-2b79693a5c2e"
      onChange={[MockFunction]}
      value="fugiat"
    >
      <option
        key="0-explicabo"
        value="explicabo"
      >
        explicabo
      </option>
      <option
        key="1-iste"
        value="iste"
      >
        iste
      </option>
      <option
        key="2-quia"
        value="quia"
      >
        quia
      </option>
      <option
        key="3-fugiat"
        value="fugiat"
      >
        fugiat
      </option>
      <option
        key="4-ducimus"
        value="ducimus"
      >
        ducimus
      </option>
      <option
        key="5-quia"
        value="quia"
      >
        quia
      </option>
      <option
        key="6-sit"
        value="sit"
      >
        sit
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 24`] = `
<div
  className="reiciendis laborum enim molestias perferendis doloremque qui"
>
  <div>
    <select
      onChange={[MockFunction]}
      value="praesentium"
    >
      <option
        key="0-praesentium"
        value="praesentium"
      >
        praesentium
      </option>
      <option
        key="1-quaerat"
        value="quaerat"
      >
        quaerat
      </option>
      <option
        key="2-laborum"
        value="laborum"
      >
        laborum
      </option>
      <option
        key="3-fugit"
        value="fugit"
      >
        fugit
      </option>
      <option
        key="4-est"
        value="est"
      >
        est
      </option>
      <option
        key="5-voluptatem"
        value="voluptatem"
      >
        voluptatem
      </option>
      <option
        key="6-et"
        value="et"
      >
        et
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 25`] = `
<div>
  <div>
    <select
      id="62f15fad-5cb3-4995-9956-bd745eed1cd2"
      onChange={[MockFunction]}
    >
      <option
        key="0-quam"
        value="quam"
      >
        quam
      </option>
      <option
        key="1-nesciunt"
        value="nesciunt"
      >
        nesciunt
      </option>
      <option
        key="2-quis"
        value="quis"
      >
        quis
      </option>
      <option
        key="3-enim"
        value="enim"
      >
        enim
      </option>
      <option
        key="4-aut"
        value="aut"
      >
        aut
      </option>
      <option
        key="5-recusandae"
        value="recusandae"
      >
        recusandae
      </option>
      <option
        key="6-doloremque"
        value="doloremque"
      >
        doloremque
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 26`] = `
<div>
  <Label>
    debitis eos rerum doloribus voluptas aliquid optio
  </Label>
  <div>
    <select
      name="f4999528-e143-4053-9bd4-b10a151b416c"
      onChange={[MockFunction]}
    >
      <option
        key="0-neque"
        value="neque"
      >
        neque
      </option>
      <option
        key="1-tempore"
        value="tempore"
      >
        tempore
      </option>
      <option
        key="2-ipsa"
        value="ipsa"
      >
        ipsa
      </option>
      <option
        key="3-perspiciatis"
        value="perspiciatis"
      >
        perspiciatis
      </option>
      <option
        key="4-illo"
        value="illo"
      >
        illo
      </option>
      <option
        key="5-laudantium"
        value="laudantium"
      >
        laudantium
      </option>
      <option
        key="6-sed"
        value="sed"
      >
        sed
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 27`] = `
<div>
  <Label
    htmlFor="3a1e59ee-4a03-4fdd-90d3-07884844b68f"
  >
    mollitia tempore delectus eos sunt optio iusto
    *
  </Label>
  <div>
    <select
      id="3a1e59ee-4a03-4fdd-90d3-07884844b68f"
      name="cefa95b7-c451-4cf6-abcf-3a83a75e2046"
      onChange={[MockFunction]}
    >
      <option
        key="0-dolores"
        value="dolores"
      >
        dolores
      </option>
      <option
        key="1-aut"
        value="aut"
      >
        aut
      </option>
      <option
        key="2-dolor"
        value="dolor"
      >
        dolor
      </option>
      <option
        key="3-soluta"
        value="soluta"
      >
        soluta
      </option>
      <option
        key="4-aspernatur"
        value="aspernatur"
      >
        aspernatur
      </option>
      <option
        key="5-beatae"
        value="beatae"
      >
        beatae
      </option>
      <option
        key="6-qui"
        value="qui"
      >
        qui
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 28`] = `
<div
  className="quidem molestias repudiandae mollitia minus error placeat"
>
  <div>
    <select
      id="ca91d3fa-ee06-40f5-a38c-0050a617f31c"
      name="a5202c2d-4857-4b3c-aede-04b5ec714923"
      onChange={[MockFunction]}
    >
      <option
        key="0-libero"
        value="libero"
      >
        libero
      </option>
      <option
        key="1-esse"
        value="esse"
      >
        esse
      </option>
      <option
        key="2-inventore"
        value="inventore"
      >
        inventore
      </option>
      <option
        key="3-enim"
        value="enim"
      >
        enim
      </option>
      <option
        key="4-necessitatibus"
        value="necessitatibus"
      >
        necessitatibus
      </option>
      <option
        key="5-nihil"
        value="nihil"
      >
        nihil
      </option>
      <option
        key="6-autem"
        value="autem"
      >
        autem
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 29`] = `
<div
  className="amet suscipit sed est corporis est quos"
>
  <div>
    <select
      id="60ac57a2-6635-4633-b776-2072fed0deab"
      name="762bfca6-954f-4da8-a403-e33c3ea72b22"
      onChange={[MockFunction]}
      value="accusantium"
    >
      <option
        key="0-perspiciatis"
        value="perspiciatis"
      >
        perspiciatis
      </option>
      <option
        key="1-accusantium"
        value="accusantium"
      >
        accusantium
      </option>
      <option
        key="2-ducimus"
        value="ducimus"
      >
        ducimus
      </option>
      <option
        key="3-possimus"
        value="possimus"
      >
        possimus
      </option>
      <option
        key="4-omnis"
        value="omnis"
      >
        omnis
      </option>
      <option
        key="5-accusamus"
        value="accusamus"
      >
        accusamus
      </option>
      <option
        key="6-dolorum"
        value="dolorum"
      >
        dolorum
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 30`] = `
<div>
  <div>
    <select
      name="af5e0813-deba-42de-bf18-1ca66d0c46cb"
      onChange={[MockFunction]}
      value="assumenda"
    >
      <option
        key="0-assumenda"
        value="assumenda"
      >
        assumenda
      </option>
      <option
        key="1-rerum"
        value="rerum"
      >
        rerum
      </option>
      <option
        key="2-officiis"
        value="officiis"
      >
        officiis
      </option>
      <option
        key="3-cumque"
        value="cumque"
      >
        cumque
      </option>
      <option
        key="4-dolore"
        value="dolore"
      >
        dolore
      </option>
      <option
        key="5-quidem"
        value="quidem"
      >
        quidem
      </option>
      <option
        key="6-sunt"
        value="sunt"
      >
        sunt
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 31`] = `
<div
  className="est facilis illum id debitis ut sint"
>
  <div>
    <select
      id="0cb84201-b118-455f-9938-447c203fbd6b"
      onChange={[MockFunction]}
    >
      <option
        key="0-sunt"
        value="sunt"
      >
        sunt
      </option>
      <option
        key="1-porro"
        value="porro"
      >
        porro
      </option>
      <option
        key="2-et"
        value="et"
      >
        et
      </option>
      <option
        key="3-sed"
        value="sed"
      >
        sed
      </option>
      <option
        key="4-aut"
        value="aut"
      >
        aut
      </option>
      <option
        key="5-quia"
        value="quia"
      >
        quia
      </option>
      <option
        key="6-explicabo"
        value="explicabo"
      >
        explicabo
      </option>
    </select>
  </div>
</div>
`;

exports[`<Select /> matches snapshot 32`] = `
<div>
  <div>
    <select
      id="c69e7e7b-983b-410f-80b0-c68d60981224"
      onChange={[MockFunction]}
    >
      <option
        key="0-fugiat"
        value="fugiat"
      >
        fugiat
      </option>
      <option
        key="1-vero"
        value="vero"
      >
        vero
      </option>
      <option
        key="2-corrupti"
        value="corrupti"
      >
        corrupti
      </option>
      <option
        key="3-nihil"
        value="nihil"
      >
        nihil
      </option>
      <option
        key="4-est"
        value="est"
      >
        est
      </option>
      <option
        key="5-maxime"
        value="maxime"
      >
        maxime
      </option>
      <option
        key="6-vitae"
        value="vitae"
      >
        vitae
      </option>
    </select>
  </div>
</div>
`;
